sw/device/tock/chips/opentitan_common/src/uart.rs

```diff
@@ -220,23 +132,40 @@ impl Uart<'a> {
     }
 
     pub fn handle_interrupt(&self) {
-        let regs = self.registers;
-        let intrs = regs.intr_state.extract();
-
-        if intrs.is_set(intr::tx_empty) {
-            self.disable_tx_interrupt();
-            self.tx_progress();
-        } else if intrs.is_set(intr::rx_watermark) {
-            self.disable_rx_interrupt();
-            // TODO: real RX processing
+        let mut interrupt_context: u32 = 0;
+        match self.interrupts_disable() {
+            Ok(v) => interrupt_context = v,
+            Err(e) => (), // TODO - return error
+        }
```

Comment:
I don't think this is needed at the moment, as we don't actually service interrupts when they occur. However, I think nothing prevents chip from servicing interrupts as they occur (not sure if there are any use-cases for this behaviour). Even if the interrupts are serviced at the point when they occur, I don't think for a system with PLIC and a single HART this would be an issue.

Bottom line, I don't think there is need for `interrupts_disable` and `interrupts_restore` providing we use RISC-V architecture, and PLIC which is  a part of it.

Reply:
It is true that Tock interrupts are not intended to be serviced as they occur. Nonetheless, I was concerned that removing `interrupts_disable` and `interrupts_restore` seems not right as every other UART driver in Tock still disables some set of interrupts while handling UART interrupts. After further investigation I could not figure out any reason why those steps were necessary, so I asked on Slack, and it appears they are likely included throughout Tock as a consequence of porting the original drivers from C and then subsequent code reuse, but are not in fact necessary.

So I agree that there seems to be no need for `interrupts_disable()` and `interrupts_restore()` unless there is a need to disable interrupts from uart reception for some reason.

Reply:
Oh, thanks @hudson-ayers . That was only my assumption, it is great that you could confirm it!

Reply:
I have removed disable, restore functionality.